sphinx/environment/adapters/toctree.py

```diff
@@ -162,6 +163,10 @@ def _entries_from_toctree(toctreenode: addnodes.toctree, parents: List[str],
                         logger.warning(__('toctree contains reference to document %r that '
                                           'doesn\'t have a title: no link will be generated'),
                                        ref, location=toctreenode)
+                    if not included(self.env.doc2path(ref, False)):
```

Comment:
I guess this block should be moved into the `except KeyError` block. Is my understanding correct?